lib/poetry.js

```diff
@@ -21,12 +21,27 @@ async function pyprojectTomlToRequirements(modulePath, pluginInstance) {
     generateRequirementsProgress = progress.get(
       'python-generate-requirements-toml'
     );
-    generateRequirementsProgress.update(
-      'Generating requirements.txt from "pyproject.toml"'
-    );
-    log.info('Generating requirements.txt from "pyproject.toml"');
+  }
+
+  const emitMsg = (msg) => {
+    if (generateRequirementsProgress) {
+      generateRequirementsProgress.update(msg);
+      log.info(msg);
+    } else {
+      serverless.cli.log(msg);
+    }
+  };
+
+  if (fs.existsSync('poetry.lock')) {
```

Comment:
Let's hide this check behind the `requirePoetryLockFile`, there's no need to check if for users that don't even use this flag

Reply:
There are two different accurate logging messages with that check though. This is the thing that brought me to open this PR: When I started working with the serverless-python-requirements plugin and saw the former message, I was unsure what it was actually doing so I had to open the code to verify.

That being said, tell me what you want me to do. :)

Reply:
Hey @FinchPowers, I'm not sure I understand. Why do you think it makes sense to check it for each user, even the ones that don't use the introduced flag? Maybe I'm missing something here

Reply:
It's to provide accurate logging. Let's say the flag does not exist, that would still leave this

```
if (fs.existsSync('poetry.lock')) {
  emitMsg('Generating requirements.txt from poetry.lock');
} else {
  emitMsg('Generating poetry.lock and requirements.txt from pyproject.toml');
}
```

Reply:
@pgrzesik Standing by: Tell me what you would like me to do. :)

Reply:
Hey @FinchPowers 👋 Sorry for not responding sooner, I was on a short vacation for pretty much the whole last week. 

That makes a lot of sense, thanks for the explanation 👍